libindy/src/services/pool/state_proof/mod.rs

```diff
@@ -533,6 +526,7 @@ fn _verify_proof_signature(signature: &str,
 }
 
 fn _parse_reply_for_proof_value(json_msg: &SJsonValue, data: Option<&str>, parsed_data: &SJsonValue, xtype: &str, sp_key: &[u8]) -> Result<Option<String>, String> {
+    use sha2::digest::{FixedOutput, Input};
```

Comment:
Should this `use` be moved to the top of the file?

Reply:
Getting this to compile was astonishingly complicated. 

At least when I upgraded sha3 after getting sha2 to compile after its update.

Maybe it is working now because I removed dependencies for generic-array and digest from Cargo.toml. libindy is not using them directly but only through dependencies.
rustc seems to be of different opinion than me which digest (digest, sha2::digest or sha3::digest) and which generic-array (generic-array, sha::generic-array or sha3::generic-array) must be used. The same for Input and FixedOutput.

That is why I have put the use-statements close to where the methods/types/traits are used.

I am traveling today and can't try it quickly.

ps: have you seen the issue I created regarding dependabot? https://jira.hyperledger.org/browse/IS-1296 That would create PRs for new versions of dependencies automatically. Although, as this example sha2/3 shows the PR would not compile, but at least raise awareness of a new version.
pps: BTW rusqlite made a big leap version-wise.